openssl/0002-Support-MSYS2.patch

```diff
@@ -0,0 +1,229 @@
+ $target{cxxflags}//=$target{cflags} if $target{CXX};
+ $target{exe_extension}="";
+ $target{exe_extension}=".exe" if ($config{target} eq "DJGPP"
+-                                  || $config{target} =~ /^(?:Cygwin|mingw)/);
++                                  || $config{target} =~ /^(?:Cygwin|Msys|mingw)/);
+ $target{exe_extension}=".pm"  if ($config{target} =~ /vos/);
+ 
+ ($target{shared_extension_simple}=$target{shared_extension})
+@@ -1132,7 +1132,7 @@ $target{exe_extension}=".pm"  if ($config{target} =~ /vos/);
+     unless defined($target{shared_extension_simple});
+ $target{dso_extension}//=$target{shared_extension_simple};
+ ($target{shared_import_extension}=$target{shared_extension_simple}.".a")
+-    if ($config{target} =~ /^(?:Cygwin|mingw)/);
++    if ($config{target} =~ /^(?:Cygwin|Msys|mingw)/);
+ 
+ # Fill %config with values from %user, and in case those are undefined or
+ # empty, use values from %target (acting as a default).
+diff --git a/apps/build.info b/apps/build.info
+index 751d8da..4c27432 100644
+--- a/apps/build.info
++++ b/apps/build.info
+@@ -22,7 +22,7 @@ IF[{- !$disabled{apps} -}]
+   INCLUDE[openssl]=.. ../include
+   DEPEND[openssl]=libapps.a ../libssl
+ 
+-IF[{- $config{target} =~ /^(?:Cygwin|mingw|VC-)/ -}]
++IF[{- $config{target} =~ /^(?:Cygwin|Msys|mingw|VC-)/ -}]
+   GENERATE[openssl.rc]=../util/mkrc.pl openssl
+   SOURCE[openssl]=openssl.rc
+ ENDIF
+diff --git a/build.info b/build.info
+index 3dda4e8..25e198a 100644
+--- a/build.info
++++ b/build.info
+@@ -71,7 +71,7 @@ ELSIF[{- $config{target} =~ /^vms/ -}]
+   SHARED_SOURCE[libssl]=libssl.opt
+ ENDIF
+ 
+-IF[{- $config{target} =~ /^(?:Cygwin|mingw|VC-)/ -}]
++IF[{- $config{target} =~ /^(?:Cygwin|Msys|mingw|VC-)/ -}]
+   GENERATE[libcrypto.rc]=util/mkrc.pl libcrypto
+   GENERATE[libssl.rc]=util/mkrc.pl libssl
+ 
+@@ -82,6 +82,9 @@ ENDIF
+ IF[{- $config{target} =~ /^Cygwin/ -}]
+  SHARED_NAME[libcrypto]=cygcrypto-{- $sover_filename -}
+  SHARED_NAME[libssl]=cygssl-{- $sover_filename -}
++ELSIF[{- $config{target} =~ /^Msys/ -}]
++ SHARED_NAME[libcrypto]=msys-crypto-{- $sover_filename -}
++ SHARED_NAME[libssl]=msys-ssl-{- $sover_filename -}
+ ELSIF[{- $config{target} =~ /^mingw/ -}]
+  SHARED_NAME[libcrypto]=libcrypto-{- $sover_filename -}{- $config{target} eq "mingw64" ? "-x64" : "" -}
+  SHARED_NAME[libssl]=libssl-{- $sover_filename -}{- $config{target} eq "mingw64" ? "-x64" : "" -}
+diff --git a/config b/config
+index b8adf34..02f3571 100755
+--- a/config
++++ b/config
+@@ -331,6 +331,10 @@ case "${SYSTEM}:${RELEASE}:${VERSION}:${MACHINE}" in
+ 	echo "${MACHINE}-pc-cygwin"; exit 0
+ 	;;
+ 
++    MSYS*)
++	echo "${MACHINE}-pc-msys"; exit 0
++	;;
++
+     vxworks*)
+        echo "${MACHINE}-whatever-vxworks"; exit 0;
+        ;;
+@@ -815,6 +819,7 @@ case "$GUESSOS" in
+   # these are all covered by the catchall below
+   i[3456]86-*-cygwin) OUT="Cygwin-x86" ;;
+   *-*-cygwin) OUT="Cygwin-${MACHINE}" ;;
++  *-*-msys) OUT="Msys" ;;
```

Comment:
Lets do like cygwin - ````OUT="Msys-${MACHINE}"````

Reply:
Could you please do a one-pass review instead of single comments? Thanks so much.

Reply:
This is the last that I found

Reply:
Okay.